source/components/AblyTask/utilities.brs

```diff
@@ -86,9 +86,130 @@ end function
+    method: "GET"
+    queries: Invalid
+  }
+
+  ' Override the default options with the incoming options
+  options.append(requestOptions)
+
+  if isNonEmptyAA(options.queries) then url = appendQueriesToUri(url, options.queries)
+
+  url = urlProxy(url)
+  messagePort = createObject("roMessagePort")
+  transferObject = createNewUrlTransfer(messagePort)
+  transferObject.setUrl(url)
+
+  ' Make sure the Url was correctly set.... This can fail due to bad encoding...
+  if url = transferObject.getUrl() then
+    if isNonEmptyString(options.certificateFile) then
+      transferObject.setCertificatesFile(options.certificateFile)
+      transferObject.initClientCertificates()
+    end if
+
+    if isNonEmptyAA(options.headers) then transferObject.setHeaders(options.headers)
+
+    if isNonEmptyAA(options.cookies) then
+      transferObject.enableCookies()
+      transferObject.addCookies(options.cookies)
+    else
+      transferObject.clearCookies()
+    end if
+
+    if isNonEmptyString(options.method) then transferObject.setRequest(options.method)
+
+    body = options.body
+    if isAA(body) OR isArray(body) then
+      ok = transferObject.asyncPostFromString(formatJson(body))
+    else if isString(body) then
+      ok = transferObject.asyncPostFromString(body)
+    else
+      ok = transferObject.asyncGetToString()
+    end if
+  else
+    ok = false
+  end if
+
+  if NOT ok then
+    return {
+      "ok": ok
+      "code": 0
+      "body": Invalid
+      "headers": {}
+      "messageDetails": ""
+      "url": url
+    }
+  else
+    while true
+      message = getMessage(messagePort)
+      if message <> Invalid then
+        if isUrlEvent(message) then return processResponse(message, transferObject)
+      end if
+    end while
+  end if
+end function
+
+function processResponse(message as Object, transferObject as Object) as Object
+  code = message.getResponseCode()
+  body = message.getString()
+  headers = message.getResponseHeaders()
+  requestUrl = transferObject.getUrl()
+  ' Was there a curl error?
+  ok = code >= 0
+
+  if ok AND isNonEmptyString(body) then
+    contentType = ""
+    if isNonEmptyAA(headers) AND isNonEmptyString(headers["content-type"]) then contentType = headers["content-type"]
+    if stringIncludes(contentType, "application/json") then
+      bodyAsObject = parseJson(body)
+      if isNotInvalid(bodyAsObject) then
+        body = bodyAsObject
+      end if
+    end if
+  else
+    body = Invalid
+  end if
+
+  return {
+    "ok": ok
+    "code": message.getResponseCode()
+    "body": body
+    "headers": headers
+    "messageDetails": message.getFailureReason()
+    "url": requestUrl
+  }
+end function
+
+function createNewUrlTransfer(messagePort as Object) as Object
+  transferObject = createObject("roUrlTransfer")
+  transferObject.setPort(messagePort)
+  transferObject.enableEncodings(true)
+  transferObject.retainBodyOnError(true)
+  ' transferObject.setMinimumTransferRate(2147483647, m.timeout)
```

Comment:
Looks like this line is commented out but without any surrounding meta-commentary to explain why.

Could you either add some commentary or remove it.

Reply:
I think I can remove it. Was left in from the POC days and we don't really need a hard timeout I don't think. If we need it later I can add it back.

Reply:
882a3d9